### PR TITLE
Add enable/disable system for testplugins

### PR DESCRIPTION
**Pull request by ImMorpheus**
```
/test enable|disable <plugin1> <plugin2>
/test status
```
TestPlugins are disabled by default.

### CODE REVIEW

File: testplugins/src/main/java/org/spongepowered/test/CopyPasta.java [hunk 1]
```diff
@@ -225,23 +227,37 @@ public void onGamePreInitialization(GamePreInitializationEvent event) {
...
-    public void onInteract(InteractBlockEvent.Secondary.MainHand event, @Root Player player) {
-        Optional<ItemStack> item = player.getItemInHand(HandTypes.MAIN_HAND);
-        if (item.isPresent() && item.get().getType().equals(ItemTypes.WOODEN_AXE) && event.getTargetBlock() != BlockSnapshot.NONE) {
-            get(player).setPos2(event.getTargetBlock().getPosition());
-            player.sendMessage(Text.of(TextColors.LIGHT_PURPLE, "Position 2 set to " + event.getTargetBlock().getPosition()));
-            event.setCancelled(true);
-        }
+    @Override
+    public void disable(CommandSource src) {
+        Sponge.getEventManager().unregisterListeners(this.plugin);
```
Comment (gabizou) [inline testplugins/src/main/java/org/spongepowered/test/CopyPasta.java]: Could this be streamlined/default implemented so only the plugins that have more than just the listeners will need to override?

**Review by gabizou — CHANGES_REQUESTED** [review]

File: testplugins/src/main/java/org/spongepowered/test/CopyPasta.java (hunk 1 above)
Comment (ImMorpheus) [reply testplugins/src/main/java/org/spongepowered/test/CopyPasta.java]: I would need another method to retrieve the class handling the listeners to default implement this. 
Is it worth adding more methods ?

Comment (gabizou) [reply testplugins/src/main/java/org/spongepowered/test/CopyPasta.java]: I'd say it's worth adding a method returning an object as "listener".

File: testplugins/src/main/java/org/spongepowered/test/MainTestPlugin.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                                        Sponge.getServer().getBroadcastChannel().send(Text.of(pc.getId(), " enabled"));
+                                        ((LoadableModule) plugin).enable(src);
+                                    } else {
+                                        Sponge.getServer().getBroadcastChannel().send(Text.of(pc.getId(), " is not a LoadableModule"));
+                                    }
+                                });
+                            }
+                            return CommandResult.success();
+                        }))
+                        .build(), "enableplugins");
```
Comment (ImMorpheus) [inline testplugins/src/main/java/org/spongepowered/test/MainTestPlugin.java]: `ep` and `dp`: shorter alias, faster to type 
`test enable|disable <plugin1> <plugin2>`: cleaner

Thoughts ?

Comment (phit) [reply testplugins/src/main/java/org/spongepowered/test/MainTestPlugin.java]: im fine with dropping the short alias, if you can make `/test` without any arguments a quick toggle to enable/disable

Comment (ImMorpheus) [reply testplugins/src/main/java/org/spongepowered/test/MainTestPlugin.java]: I would need to track the status (enabled/disabled) of individual plugins then. That would become uglier than the alias IMO.

Unless you want it to be just a global switch.

Comment (gabizou) [reply testplugins/src/main/java/org/spongepowered/test/MainTestPlugin.java]: I don't see why the test plugins suite couldn't have an enabled/disabled container, would be rather interesting to see as a centralized debugging spot to visualize what plugins are enabled in the event of some listeners being active or inactive. Granted, a majority of these plugins are for testing functionality, and/or bug fixes, so it's not common to have this sort of interaction being preformed. My two cents are to have a global container (specific to the test plugins of course) to keep track of that.

**Review by gabizou — APPROVED** [review]

**Comment by gabizou** [comment]
Just maybe squash commits prior to merging.

**Pull request merged**